File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        arr = {}
        for i in range( len(nums)):
            x = target - nums[i]
            if x in arr:
                return [arr[x], i]
            else:
                arr[nums[i]] = i
        return [0,0]

File: Data Structures & Algorithms/target-sum/submission-0.cpp
class Solution {
public:
    int findTargetSumWays(vector<int>& nums, int target) {
        return dfs(nums, 0, target);
    }

    int dfs(vector<int>& nums, int i, int target){
        if(i == nums.size()){
            return target == 0 ? 1 : 0;
        }
        return dfs(nums, i + 1, target - nums[i]) + dfs(nums, i + 1, target + nums[i]); 
    }

};
